src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -107,43 +235,42 @@ def llm_classify(
     if generation_info := model.verbose_generation_info():
         printif(verbose, generation_info)
 
-    # Wrap the loop in a try / catch so that we can still return a dataframe
-    # even if the process is interrupted
-    try:
-        for index, prompt in enumerate(
-            tqdm(prompts, bar_format=get_tqdm_progress_bar_formatter("llm_classify"))
-        ):
-            with set_verbosity(model, verbose) as verbose_model:
-                response = verbose_model(prompt, instruction=system_instruction, **model_kwargs)
-            if not use_openai_function_call:
-                if provide_explanation:
-                    unrailed_label, explanation = (
-                        eval_template.extract_label_from_explanation(response),
-                        response,
-                    )
-                    printif(
-                        verbose and unrailed_label == NOT_PARSABLE,
-                        f"- Could not parse {repr(response)}",
-                    )
-                else:
-                    unrailed_label = response
-                    explanation = None
+    async def _run_llm_classification(prompt: str) -> Tuple[str, Optional[str]]:
+        with set_verbosity(model, verbose) as verbose_model:
+            response = await verbose_model._async_generate(
+                prompt, instruction=system_instruction, **model_kwargs
+            )
+        if not use_openai_function_call:
+            if provide_explanation:
+                unrailed_label, explanation = (
+                    eval_template.extract_label_from_explanation(response),
+                    response,
+                )
+                printif(
+                    verbose and unrailed_label == NOT_PARSABLE,
+                    f"- Could not parse {repr(response)}",
+                )
```

Comment:
This snippet was already present, but I think we can remove. The `_snap_to_rail` function should handle logging in the case where the LLM output is not parseable.